Src/WitsmlExplorer.Frontend/components/Sidebar/ServerManager.tsx

```diff
@@ -96,17 +99,27 @@ const ServerManager = (): React.ReactElement => {
         abortController.abort();
       };
     }
-  }, [isAuthenticated]);
+  }, [isAuthenticated, hasFetchedServers]);
 
   const onSelectItem = async (server: Server) => {
-    const action: SelectServerAction = { type: NavigationType.SelectServer, payload: { server } };
+    if (server.id === selectedServer?.id) {
+      await CredentialsService.deauthorize();
+      signOut();
```

Comment:
The signOut() function from the MsalAuthProvider is used to log out of Azure from the whole browser and should thus not be used here. We currently do not have any functionality to log out of an individual server, so for now it should be sufficient to adjust the state in the frontend to behave as if we are not logged in.

Reply:
updated!